src/core/library.c

```diff
@@ -505,6 +505,28 @@ QuicLibApplyLoadBalancingSetting(
         MsQuicLib.CidTotalLength);
 }
 
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+QuicLibrarySumPerfCounters(
+    _Out_writes_bytes_(BufferLength) uint8_t* Buffer,
+    _In_ uint32_t BufferLength
+    )
+{
+    const uint32_t CountersPerBuffer = BufferLength / sizeof(int64_t);
+    int64_t* const Counters = (int64_t*)Buffer;
+    memcpy(Buffer, MsQuicLib.PerProc[0].PerfCounters, BufferLength);
+
+    for (uint32_t ProcIndex = 1; ProcIndex < MsQuicLib.ProcessorCount; ++ProcIndex) {
+        for (uint32_t CounterIndex = 0; CounterIndex < CountersPerBuffer; ++CounterIndex) {
+            Counters[CounterIndex] += MsQuicLib.PerProc[ProcIndex].PerfCounters[CounterIndex];
+
+            if (Counters[CounterIndex] < 0) {
+                Counters[CounterIndex] = 0;
+            }
```

Comment:
You need to do this only after summing them all up.

Reply:
I was trying to save another loop over the PerProc objects.
If I invert the loops so instead the outer loop is over the counters, and the inner loop is over the PerProc, I could do that. But I think that approach would hurt cache locality. Do you have an opinion against a second loop?

Reply:
Just make a second loop.